symfony/plugins/orangehrmPimPlugin/lib/service/EmployeeSubscriptionService.php

```diff
@@ -0,0 +1,146 @@
+ * Boston, MA  02110-1301, USA
+ */
+
+class EmployeeSubscriptionService extends BaseService {
+
+    /**
+     * @var sysConf
+     */
+    private $sysConf = null;
+
+    /**
+     * @param int    $empId
+     * @param string $name
+     * @param string $email
+     * @param string $instanceId
+     *
+     * @return bool
+     * @throws Exception
+     */
+    public function subscribe(int $empId, string $name, string $email): bool {
+        $employeeSubscription = $this->getEmployeeSubscription($empId);
+        $employeeSubscription->setEmployeeId($empId);
+        $employeeSubscription->setStatus(1);
+        $employeeSubscription->setCreatedAt(date('Y-m-d'));
+        $employeeSubscription->save();
+
+        //send info into remote server
+        $ch = curl_init();
+        curl_setopt($ch, CURLOPT_URL, $this->getSysConf()->getRegistrationUrl() . '/subscribe.php');
+        curl_setopt($ch, CURLOPT_POST, 1);
+
+        $data = "name=" . $name
+            . "&email=" . $email
+            . "&instance_id=" . $this->getInstanceId();
+
+        curl_setopt($ch, CURLOPT_POSTFIELDS, $data);
+        curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
+        curl_exec($ch);
+        $http_status = curl_getinfo($ch, CURLINFO_HTTP_CODE);
+        curl_close($ch);
+        if (!($http_status === 200)) {
+            return false;
+        }
+        else {
+
+            return true;
+        }
+    }
+
+    /**
+     * Unsubscribe employee from the mail service
+     *
+     * @param int $empId
+     *
+     * @return bool
+     */
+    public function unsubscribe(int $empId): bool {
+        $employeeSubscription = $this->getEmployeeSubscription($empId);
+        $employeeSubscription->setEmployeeid($empId);
+        $employeeSubscription->setStatus(0);
+        $employeeSubscription->save();
+
+        return true;
+        //@todo send api request to external service
+    }
+
+    /**
+     * @param int $empId
+     *
+     * @return EmployeeSubscription
+     */
+    private function getEmployeeSubscription(int $empId): EmployeeSubscription {
+        $q = Doctrine_Query::create()
+            ->from('EmployeeSubscription')
+            ->where('employee_id = ?', $empId);
+
+        $employeeSubscription = $q->fetchOne();
+        if (empty($employeeSubscription)) {
+            $employeeSubscription = new EmployeeSubscription();
+        }
+
+        return $employeeSubscription;
+    }
+
+    /**
+     * Check emloyee already subscribed
+     *
+     * @param int $empId
+     *
+     * @return bool
+     * @throws Doctrine_Query_Exception
+     */
+    public function isSubscribed(int $empId): bool {
+        $q = Doctrine_Query::create()
+            ->from('EmployeeSubscription')
+            ->where('employee_id = ?', $empId);
+
+        $employeeSubscription = $q->fetchOne();
+
+        if (!empty($employeeSubscription)) {
+            return true;
+        }
+
+        return false;
```

Comment:
better to replace with following code.
```
$employeeSubscription = $this->getEmployeeSubscription($empId);
return $employeeSubscription->exists();
```
aren't we consider subscription status?

Reply:
planing to delete the record when unsubscribe